src/coord/src/coord.rs

```diff
@@ -2324,7 +2313,7 @@ where
                 self.sequence_send_diffs(id, rows, affected_rows, MutationKind::Insert)
                     .await
             }
-            other => bail!("INSERT statement expected values, found {:?}", other),
+            _ => bail!("INSERT source cannot depend on other relations"),
```

Comment:
I'm not sure I follow this edit; is it bad or complicated to surface the defective result?

Reply:
It's hideous yeah, because it prints out a query AST. But I'll defer to another commit. This change is no longer here.

Reply:
Here's an example, just for kicks:

```
ERROR:  INSERT statement expected values, found Project { input: Map { input: Union { base: Project { input: Get { id: Global(User(82)), typ: RelationType { column_types: [ColumnType { nullable: true, scalar_type: Int32 }, ColumnType { nullable: false, scalar_type: String }], keys: [] } }, outputs: [0] }, inputs: [Map { input: Union { base: Negate { input: Reduce { input: Get { id: Global(User(82)), typ: RelationType { column_types: [ColumnType { nullable: true, scalar_type: Int32 }, ColumnType { nullable: false, scalar_type: String }], keys: [] } }, group_key: [], aggregates: [], monotonic: false } }, inputs: [Constant { rows: [(Row{[]}, 1)], typ: RelationType { column_types: [], keys: [] } }] }, scalars: [Literal(Ok(Row{[Null]}), ColumnType { nullable: true, scalar_type: Int32 })] }] }, scalars: [Literal(Ok(Row{[Int32(1)]}), ColumnType { nullable: false, scalar_type: Int32 }), CallUnary { func: CastInt32ToString, expr: Column(0) }] }, outputs: [1, 2] }
```

Reply:
You made the right call!